### PR TITLE
[Enhancement] Update async intrinsic handling in inject_fence_proxy

### LINKED ISSUE #1002: [Bug] Generating useless `tl::fence_proxy_async()`
```python
import torch
import tilelang
from tilelang import language as T


@tilelang.jit(
    pass_configs={
        tilelang.PassConfigKey.TL_DISABLE_WARP_SPECIALIZED: True,
        tilelang.PassConfigKey.TL_DISABLE_TMA_LOWER: True,
    },
)
def get_buggy_kernel(hidden):
    num_tokens = T.symbolic('num_tokens')

    @T.prim_func
    def buggy_kernel(x: T.Tensor[(num_tokens, hidden), 'float']):
        with T.Kernel(num_tokens, threads=128) as pid:
            smem = T.alloc_shared((hidden, ), dtype='float')
            T.copy(x[pid, :], smem)
            T.cumsum(smem)

    return buggy_kernel


if __name__ == '__main__':
    kernel = get_buggy_kernel(128)
    print(kernel.get_kernel_source())

    x = torch.zeros((1, 128), dtype=torch.float, device='cuda')
    kernel(x)
```

As titled, generated CUDA:
```cuda
extern "C" __global__ void __launch_bounds__(128, 1) buggy_kernel_kernel(float* __restrict__ x, int num_tokens) {
  extern __shared__ __align__(1024) float smem[];
  #pragma unroll
  for (int i = 0; i < 1; ++i) {
    smem[((int)threadIdx.x)] = x[((((int64_t)((int)blockIdx.x)) * (int64_t)128) + ((int64_t)((int)threadIdx.x)))];
  }
  tl::fence_proxy_async();
  __syncthreads();
  tl::CumSum1D<128, false>::run((&(smem[0])), (&(smem[0])), 128);
}
```

### PR DESCRIPTION
**Pull request by Rachmanino**
Fix #1002
* Added support for wgmma async intrinsics in IsAsyncIntrinsic function.
* Changed handling of unknown externs to treat them as Generic instead of Async, improving accuracy in proxy kind determination.
This pull request updates the logic for identifying and handling asynchronous intrinsic calls in the `inject_fence_proxy` transformation. The main improvements are to more accurately classify certain intrinsic operations and to refine the handling of unknown external calls, reducing unnecessary conservatism.

Key changes include:

Intrinsic classification improvements:
* Added explicit recognition of `tl_gemm` and `tl_gemm_sp` as asynchronous intrinsics in `IsAsyncIntrinsic`, ensuring these are properly treated as async operations.

Extern call handling refinement:
* Changed the default handling of unknown external calls from `Async` to `Generic` in `ProxyFenceInjector`, based on the guarantee that GEMM operations are always represented as intrinsic calls and not as externs. This prevents over-insertion of fences for extern calls that are not actually asynchronous.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved compiler handling of certain matrix-multiply intrinsics: refined async classification and mapping of unknown external calls for more reliable builds.

* **Tests**
  * Updated tests to match intrinsic-call behavior changes, added debug output of transformed modules, and removed an obsolete test case.

### CODE REVIEW

**Comment by Rachmanino** [comment]
It seems that we have to fix legacy tests, where gemm is still represented as extern🤔

**Comment by LeiWang1999** [comment]
I'll take a look

**Comment by Rachmanino** [comment]
> I'll take a look

really appreciate your help
